### PR TITLE
Normalize line endings when matching recordings

### PR DESCRIPTION
**Pull request by JoshLove-msft**

### CODE REVIEW

**Review by vinagesh — APPROVED** [review]

File: sdk/core/Azure.Core.TestFramework/src/RecordSession.cs [hunk 1]
```diff
@@ -89,7 +92,14 @@ public RecordEntry Lookup(RecordEntry requestEntry, RecordMatcher matcher, Recor
...
                 return entry;
             }
+        }
 
+        private void NormalizeLineEndings(RecordEntryMessage entryMessage)
+        {
+            if (entryMessage.TryGetBodyAsText(out string text))
+            {
+                entryMessage.Body = Encoding.UTF8.GetBytes(text.Replace("\n", "\r\n"));
+            }
```
Comment (JoshLove-msft) [inline sdk/core/Azure.Core.TestFramework/src/RecordSession.cs:102]: Need to either check that environment.newline is \n or use regex to ensure \n is not preceded by \r

**Comment by pakrym** [comment]
I think this is a recording roundtripping bug, not a matching bug. We need to fix the serialization mechanism so save recording was the same as a loaded one.

**Comment by JoshLove-msft** [comment]
> I think this is a recording roundtripping bug, not a matching bug. We need to fix the serialization mechanism so save recording was the same as a loaded one.

Not sure I follow - the issue occurs because the line breaks are treated as \n on linux/max, but the recordings were done on windows, so the matching fails.

**Pull request closed without merging**